src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -1360,7 +1336,7 @@ uint64_t GetAvailablePageFile()
 //      that is in use (0 indicates no memory use and 100 indicates full memory use).
 //  available_physical - The amount of physical memory currently available, in bytes.
 //  available_page_file - The maximum amount of memory the current process can commit, in bytes.
-void GCToOSInterface::GetMemoryStatus(uint64_t restricted_limit, uint32_t* memory_load, uint64_t* available_physical, uint64_t* available_page_file)
+void GCToOSInterface::GetMemoryStatus(unsigned long long restricted_limit, uint32_t* memory_load, unsigned long long* available_physical, unsigned long long* available_page_file)
```

Comment:
What's wrong with `uint64_t` ? We prefer to use types with well-defined widths like `uint64_t`.

Reply:
on MacOS only looks like platform headers define `uint64_t` as `unsigned long long`. So without this change it fails to compile only on that platform.

Reply:
What is the error that you are seeing without this change?

This file compiles as-is macOS today. It sounds like some other change in this PR broke it. Should that other change (I do not know what it is) be fixed instead?

Reply:
Here is the build break occuring only on MacOS: https://dev.azure.com/dnceng-public/public/_build/results?buildId=50960&view=logs&jobId=d69d97fe-d5be-517f-4ef6-fd1eef0215b1&j=bc2eb499-a431-5604-42ba-6d57425296b3&t=bc68434a-36cf-57f5-eb16-8887438590a6. 

@janvorli thought there is a mismatch with what is defined in `pal_mstypes.h` and MacOS headers (which `gc_pal` is using)

Reply:
Can we fix `pal_mstypes.h` to make it match instead of compensating for the mismatch here?

Reply:
I've tried to help Manish with it yesterday and it opened a can of worms. The issue with the uint64_t here is that the definition is compiled against OS headers while the declaration is used in coreclr that is compiled against PAL headers. The OS declares uint64_t as 
`typedef unsigned long long uint64_t` while pal_mstypes.h as `typedef unsigned __int64 uint64_t` and the `__int64` is declared as `#define __int64  long`. There is a long comment on why it is done this way. I've tried to change it to `#define __int64 long long`, but it broke many things in the runtime. I've tried to fix it, but that opened another can of worms so in the end, I've resorted to the change in the signature of that GCToOSInterface method.
I am not saying it is not doable the other way, but it seemed to need too many changes.

Reply:
I've given it another try today and I got it compiling on macOS x64 / arm64 with `__int64`  defined as  `long long` and `size_t` as `long` with reasonable amount of changes in coreclr and JIT. I still need to verify / fix other targets.

Reply:
CI is finally green as of now. We can wait for your PR and merge it back which would avoid this `long long` change.

Reply:
this is now reverted.